code/datums/action.dm

```diff
@@ -588,6 +597,12 @@
 		return ..() // This means that the spell is charged but unavailable due to something else
 
 	var/alpha = 220 - 140 * progress
+	var/datum/spell_cooldown/charges/handler = S.cooldown_handler
+	if(istype(handler))
+		if(handler.charge_time > world.time)
+			alpha = 220
+		else if(handler.current_charges != 0)
+			alpha = 60
```

Comment:
This is logic that should be part of the handler itself. Else you'll get snowflake code.

Reply:
done in https://github.com/ParadiseSS13/Paradise/pull/21065/commits/e40dced0c1737cd876701bf510154cddc8afd844